File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    participantCount = {}
    for i in participant:
        if i in participantCount:
            participantCount[i] += 1
        else:
            participantCount[i] = 1
    for i in completion:
        if i in participantCount:
            participantCount[i] -= 1

    for key, value in participantCount.items():
        if value != 0:
            return key

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
//         List<String> names = new ArrayList<String>(Arrays.asList(participant));
        
//         for (String c: completion) {
//             names.remove(c);
//         }
        
//         return names.get(0);
        
        Map<String, Integer> map = new HashMap<String, Integer>();
        
        for (String p: participant) {
            map.put(p, map.getOrDefault(p, 0)+1);
        }
        
        for (String c: completion) {
            map.put(c, map.getOrDefault(c, 0)-1);
        }
        
        for (String key: map.keySet()) {
            if (map.get(key)== 1) {
                return key;
            } 
        }
        return "";
    }
}